pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -287,11 +287,14 @@ const (
 
 // VolumeOptions specifies capacity and tags for a volume.
 type VolumeOptions struct {
-	CapacityGB       int
-	Tags             map[string]string
-	PVCName          string
-	VolumeType       string
-	AvailabilityZone string
+	CapacityGB        int
+	Tags              map[string]string
+	PVCName           string
+	VolumeType        string
+	ZonePresent       bool
```

Comment:
So if I understand correctly, zone or zones == "" is invalid, and should return an error, while zone && zones == nil is valid.  Can you add unit tests for all these permutations of zone/zones == nil, "", and value?